source/lac/petsc_precondition.cc

```diff
@@ -488,32 +490,62 @@ namespace PETScWrappers
     AssertThrow (ierr == 0, ExcPETScError(ierr));
 
     if (additional_data.output_details)
-      PetscOptionsSetValue("-pc_hypre_boomeramg_print_statistics","1");
+#if PETSC_VERSION_GT(3,7,0)
```

Comment:
> > @@ -488,32 +490,62 @@ namespace PETScWrappers
> >      AssertThrow (ierr == 0, ExcPETScError(ierr));
> >
> >      if (additional_data.output_details)
> > -      PetscOptionsSetValue("-pc_hypre_boomeramg_print_statistics","1");
> > +#if PETSC_VERSION_GT(3,7,0)
>
> If the API changes with 3.7.0 and you compile the code only with >3.7.0 it
> won't work with the 3.7.0 release. Can't you do
>
> #if <=3.6.0
> do old stuff
> #else
> do new stuff
> #endif
>
>
Sadly, no we can not.

The current release version of PETSc is 3.6.3. The current development
version of PETSc is 3.6.3. Yes, you read that correctly, the version
numbers are the same!

Calling >3.7.0 now, means that PETSc release does the old stuff and PETSc
dev does the new stuff. Version 3.7.0 is unknown to PETSc and therefore he
treats the version number as infinity (smart!). Calling >3.7 after the next
PETSc release will use the 3.7.0 version or later - so our code should be
compatible forever after - assuming no API changes...   ;-)   If I am wrong
about this, our PETSc guys will listen to my screams.   :-)

It's a weird PETSc logic that SLEPc clones and there never was and probably
never will be a way around it. We can not distinguish between two versions
of a library that have the same version numbers (stupid!) unless we call
infinity.

Hence, the method I proposed earlier is in my opinion the simplest for us;
namely, change all DEAL_II_PETSC_VERSION_XXX to use native
PETSC_VERSION_XXX. After PETSc 3.7.0 is released, which will be a good
test, I will work on this change.